src/ngraph/attribute_adapter.hpp

```diff
@@ -0,0 +1,140 @@
+//*****************************************************************************
+// Copyright 2017-2019 Intel Corporation
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//*****************************************************************************
+
+#pragma once
+
+#include "ngraph/enum_names.hpp"
+#include "ngraph/type.hpp"
+
+namespace ngraph
+{
+    /// Provides a generic way to access attribute values as strings for serialization
+    class StringAdapter
+    {
+    public:
+        virtual ~StringAdapter() {}
+        static constexpr DiscreteTypeInfo type_info{"StringAdapter", 0};
+        virtual const DiscreteTypeInfo& get_type_info() const { return type_info; }
+        /// Returns the value as a string
+        virtual std::string get_string() const = 0;
+        ///
+        virtual void set_string(const std::string& value) const = 0;
+    };
+
+    class IntegerVectorAdapter
```

Comment:
Maybe we can add template VectorAdapter which will allow to work with any vectors.
Or VectorAdapter will work together other adapters for specific types.